cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedExpr.qll

```diff
@@ -2608,3 +2610,36 @@ class TranslatedConditionDeclExpr extends TranslatedNonConstantExpr {
     result = getTranslatedExpr(expr.getVariableAccess().getFullyConverted())
   }
 }
+
+class TranslatedStmtExpr extends TranslatedNonConstantExpr {
+  override StmtExpr expr;
+  
+  override final Instruction getFirstInstruction() {
+    result = getStmt().getFirstInstruction()
+  }
+  
+  override final TranslatedElement getChild(int id) {
+    id = 0 and result = getStmt()
+  }
+  
+  override Instruction getInstructionSuccessor(InstructionTag tag, EdgeKind kind) {
+    none()
+  }
+  
+  override Instruction getChildSuccessor(TranslatedElement child) {
+    result = getParent().getChildSuccessor(this)
```

Comment:
This needs `and child = getStmt()` to avoid holding for all `TranslatedElement`s in the snapshot.

Reply:
@rdmarsh2 Given that this went unnoticed, please make sure you test these changes for performance on a snapshot that uses many statement expressions. The heaviest user I know of is the Linux kernel.